routes/api/contacts.js

```diff
@@ -1,25 +1,90 @@
-const express = require('express')
+const express = require("express");
+const contacts = require("../../models/contacts");
+const { validateAddContact, validateUpdateContact } = require("../../dataValidation");
 
-const router = express.Router()
+const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const allContacts = await contacts.listContacts();
+    res.status(200).json({
+      message: "success",
+      data: { allContacts },
+    });
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const contact = await contacts.getContactById(contactId);
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+    if (contact) {
+      res.status(200).json({
+        message: "success",
+        data: { contact },
+      });
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const body = req.body;
+    const { error } = validateAddContact(body);
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+    if (error) return res.status(400).send({ message: error.details });
 
-module.exports = router
+    const allContactsData = await contacts.addContact(body);
+    res.status(201).json({
+      message: "contact added",
+      data: { allContactsData },
+    });
+  } catch (error) {
+    console.log(error);
```

Comment:
what user will receive as response if you land in `catch`?

Reply:
added answer